os/arch/arm/src/armv7-a/arm_gicv2.c

```diff
@@ -508,6 +508,43 @@ int up_prioritize_irq(int irq, int priority)
 	return -EINVAL;
 }
 
+/****************************************************************************
+ * Name: up_affinity_irq
+ *
+ * Description:
+ *   set the affinity of a given irq (SPI only) to given cpu.
+ *   Currenlty, it doesnot support assigning multiple cpus as affinity.
+ *   It can assign only one cpu per SPI.
+ *
+ * Input Parameters:
+ *   irq: irq id
+ *   cpu: cpu id
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void up_affinity_irq(int irq, int cpu)
+{
+	DEBUGASSERT(irq >= GIC_IRQ_SPI && irq < NR_IRQS && cpu >= 0 && cpu < CONFIG_SMP_NCPUS);
+
```

Comment:
Even though we have added debugassert above, we need to add if condition here to check whether irq value is within range. If irq parameter is out of range, then we dont do anything in this api. Debug assert will be usefult only if the debug macro is enabled. But when the code goes into production, then this macro is not enabled. But we must still check the valid range of the irq parameter.